### PR TITLE
INTLY-934 Add OpenShift monitoring federation

### PR DESCRIPTION
**Pull request by aidenkeating**
Federate with the Prometheus in the OpenShift Monitoring from the
middleware monitoring namespace instead of using our own instance
of Kube State Metrics.

Verification:
- Run the installer and ensure it completes
- Ensure Kube State Metrics is not in middleware-metrics
- Ensure Alerts are passing (they may be pending for a while)
- Run the uninstaller and ensure it completes

### CODE REVIEW

**Comment by aidenkeating** [comment]
@sedroche I'm going to update this PR to follow your role structure, as our PRs should probably have their logic end up in the same `config` role.

**Comment by sedroche** [comment]
:eyes:

File: evals/roles/middleware_monitoring_config/jobs/openshift_monitoring_federation.j2 [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- job_name: openshift-monitoring-federation
+  honor_labels: true
+  kubernetes_sd_configs:
+  - role: service
+    namespaces:
+      names:
+      - openshift-monitoring
```
Comment (sedroche) [inline evals/roles/middleware_monitoring_config/jobs/openshift_monitoring_federation.j2]: I think it makes sense to template this value as well

File: evals/roles/middleware_monitoring_config/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "copy ksm template"
```
Comment (sedroche) [inline evals/roles/middleware_monitoring_config/tasks/main.yml]: https://github.com/integr8ly/installation/blob/ef57460fd9eea52a0c29936e32f022eba7679b0b/evals/roles/middleware_monitoring_config/tasks/main.yml#L2-L11

Probably worth separating this into a `create_ksm_alerts` task file and including it

**Comment by sedroche** [comment]
@aidenkeating Looks good just a couple of comments.

**Comment by sedroche** [comment]
Verifying

**Comment by aidenkeating** [comment]
Closing for #386

**Pull request closed without merging**